connector/src/main/java/org.apache.flink/connector/nebula/sink/NebulaEdgeBatchTableOutputFormat.java

```diff
@@ -0,0 +1,59 @@
+package org.apache.flink.connector.nebula.sink;
+
+import java.util.Map;
+import java.util.function.Function;
+import org.apache.flink.connector.nebula.connection.NebulaGraphConnectionProvider;
+import org.apache.flink.connector.nebula.connection.NebulaMetaConnectionProvider;
+import org.apache.flink.connector.nebula.statement.EdgeExecutionOptions;
+import org.apache.flink.connector.nebula.utils.VidTypeEnum;
+import org.apache.flink.connector.nebula.utils.WriteModeEnum;
+import org.apache.flink.table.connector.sink.DynamicTableSink.DataStructureConverter;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.types.Row;
+
+public class NebulaEdgeBatchTableOutputFormat
+        extends NebulaBatchOutputFormat<RowData, EdgeExecutionOptions> {
+    private final DataStructureConverter dataStructureConverter;
+
+    public NebulaEdgeBatchTableOutputFormat(NebulaGraphConnectionProvider graphProvider,
+                                            NebulaMetaConnectionProvider metaProvider,
+                                            EdgeExecutionOptions executionOptions,
+                                            DataStructureConverter dataStructureConverter) {
+        super(graphProvider, metaProvider, executionOptions);
+        this.dataStructureConverter = dataStructureConverter;
+    }
+
+    @Override
+    protected NebulaBatchExecutor<RowData> createNebulaBatchExecutor() {
+        VidTypeEnum vidType = metaProvider.getVidType(metaClient, executionOptions.getGraphSpace());
+        Map<String, Integer> schema = metaProvider.getEdgeSchema(
+                metaClient,
+                executionOptions.getGraphSpace(),
+                executionOptions.getLabel());
+        EdgeExecutionOptions upsertOptions = executionOptions.toBuilder()
+                .setWriteMode(WriteModeEnum.INSERT)
+                .build();
+        EdgeExecutionOptions deleteOptions = executionOptions.toBuilder()
+                .setWriteMode(WriteModeEnum.DELETE)
+                .build();
+        return new NebulaTableBufferReducedExecutor(dataStructureConverter,
```

Comment:
We do not suggest the upset mode, because the performance of `upsert` is not good enough. 
Refer https://docs.nebula-graph.com.cn/3.3.0/3.ngql-guide/12.vertex-statements/3.upsert-vertex/

Reply:
Yes, we use the Nebula `INSERT` statements for both insert and update events in the data stream. What I found is that NebulaGraph will override existing vertices/edges during insert, so it works for the update event as well. (`UPSERT` can be expensive in NebulaGraph so I tried to avoid it.)

> maybe we should change the upsert variable to `insert`

Sounds good! I've renamed the variables.